sudoku_solver.py

```diff
@@ -36,20 +35,44 @@ def get_position(x: int) -> int:
     return (x // 3) * 3
 
 
-if __name__ == "__main__":
-    fake = np.empty((9,9))
-    "for every 1-9, check every row"
+def go_through_sudoku(sudoku: np.array) -> np.array:
+    """
+    Runs through the sudoku and
+    enters any valid numbers for that run through
+    """
     for i in range(9):
         for j in range(9):
             possible_values = []
-            if test_sudoku[i, j-1] == "x":
+            if sudoku[i, j] == "x":
                 for possible_value in range(1, 10):
-                    row = test_sudoku[i, :]
-                    column = test_sudoku[:, j]
+                    row = sudoku[i, :]
+                    column = sudoku[:, j]
                     i_start_pos = get_position(i)
                     j_start_pos = get_position(j)
-                    box = test_sudoku[i_start_pos: i_start_pos + 3, j_start_pos: j_start_pos + 3]
+                    box = sudoku[i_start_pos: i_start_pos +
+                                 3, j_start_pos: j_start_pos + 3]
                     if check_all_conditions(row, column, box, possible_value):
                         possible_values.append(possible_value)
-                fake[i, j] = possible_values # TODO solve numpy issue
-    print(fake)
+                if len(possible_values) == 1:
+                    sudoku[i, j] = possible_values[0]
```

Comment:
makes complete sense but would be quicker to do the previously thought of solution where it is multiple numbers until it is 1 as it would save time. I will try to tackle that today after verifying the code above